src/spoke/Spoke.sol

```diff
@@ -302,36 +302,46 @@ abstract contract Spoke is ISpoke, Multicall, NoncesKeyed, AccessManagedUpgradea
     _validateRepay(reserve);
 
     uint256 drawnIndex = reserve.hub.getAssetDrawnIndex(reserve.assetId);
-    (uint256 drawnDebtRestored, uint256 premiumDebtRayRestored) = userPosition.getDebt(drawnIndex);
-
-    (drawnDebtRestored, premiumDebtRayRestored) = _calculateRestoreAmount(
-      drawnDebtRestored,
-      premiumDebtRayRestored,
+    (uint256 drawnDebtRestored, uint256 premiumDebtRayRestored) = _calculateRestoreAmount(
+      userPosition,
+      drawnIndex,
       amount
     );
 
+    // we preview the drawn shares to be restored, should always be the same value returned by hub.restore()
+    uint256 previewRestoredShares = reserve.hub.previewRestoreByAssets(
```

Comment:
I know it might be a stretch, but what do you think of previewing it by calculating here manually (divUp), and call it `restoredShares`. Hub is immutable, and we can hardcode/duplicate logic here (we already do it  in other places)

Reply:
current Hub is immutable, but future ones might have another logic applied during the preview, I'd prefer to trust Hub return values rather than calculating locally.